src/confd/yang/infix-services.yang

```diff
@@ -36,6 +59,49 @@ module infix-services {
       type boolean;
     }
   }
+  container ssh {
+    description "Configuration for the SSH daemon";
+    leaf enabled {
+      description "Disable or enable SSH daemon";
+      type boolean;
+      must ". = 'false' or (count(../listen) > 0 and count(../hostkey) > 0)" {
+        error-message "When SSH is enabled, there must be at least one listen address configured and a private key must be set.";
+      }
+    }
+
+    leaf-list hostkey {
+      description
+        "A reference to an asymmetric key that exists in
+          the central keystore.";
+      must "not(deref(.)/../ks:public-key-format) or "
+        + "(derived-from-or-self(deref(.)/../ks:public-key-format,  'ct:ssh-public-key-format') and"
+        + "derived-from-or-self(deref(.)/../ks:private-key-format,  'ct:rsa-private-key-format'))" {
+        error-message "Only RSA hostkeys are supported";
+      }
+      type ks:asymmetric-key-ref;
+    }
+
+    list listen {
+      key name;
+      leaf name {
+        type string;
+      }
+
+      leaf address {
+        description
+          "The local IP address to listen on for incoming
+           SSH client connections.  INADDR_ANY (0.0.0.0) or
+           INADDR6_ANY (0:0:0:0:0:0:0:0 a.k.a. ::) MUST be
+           used when the server is to listen on all IPv4 or
+           IPv6 addresses, respectively.";
+        type inet:ip-address;
+      }
+      leaf port {
+        description "Local port for SSH daemon to listen to.";
```

Comment:
To listen _on_, or to listen _to_?   We should use the same preposition for this and the `address` description.